server/s3_log.h

```diff
@@ -66,16 +70,10 @@ inline const char* s3_log_get_req_id(const std::string& requestid) {
 #define s3_log(loglevel, requestid, fmt, ...)                    \
   do {                                                           \
     if (loglevel >= s3log_level) {                               \
-      char* s3_log_msg__ = nullptr;                              \
-      int s3_log_len__ =                                         \
-          asprintf(&s3_log_msg__, "[Req:%s] " fmt "\n",          \
-                   s3_log_get_req_id(requestid), ##__VA_ARGS__); \
-      if (s3_log_len__ > 0) {                                    \
-        if (s3_log_msg__[s3_log_len__ - 2] == '\n')              \
-          s3_log_msg__[s3_log_len__ - 1] = '\0';                 \
-        s3_log_msg_##loglevel(s3_log_msg__);                     \
-        free(s3_log_msg__);                                      \
```

Comment:
no need of calling free(), that great

Reply:
What happens if it does call free(), will it seg fault?

Reply:
asnprintf needs free of s3_log_msg__ as that will be doing internal allocation from heap
with snprintf() its not needed, static array is passed to it.
Mostly compiler wont allow free of non heap allocated memory

Reply:
free() has constant complexity, allocation can be heavy in cases of memory pressure.